### PR TITLE
refactor: extract api calls into client with types

### PR DESCRIPTION
**Pull request by setchy**
Refactored most uses of apiRequestAuth into a new `api/client`.  Add explicit types where missing.  Fixed semantics around Release vs ReleaseComment.  

I've named each function to be close-to that documented in https://docs.github.com/en/rest/, and included a link to the public docs where available.

### CODE REVIEW

File: src/context/App.tsx [hunk 1]
```diff
@@ -166,11 +165,8 @@ export const AppProvider = ({ children }: { children: ReactNode }) => {
 
   const validateToken = useCallback(
     async ({ token, hostname }: AuthTokenOptions) => {
-      await apiRequestAuth(
-        `${generateGitHubAPIUrl(hostname)}notifications`,
-        'HEAD',
-        token,
-      );
+      await headNotifications(hostname, token);
```
Comment (afonsojramos) [inline src/context/App.tsx:168]: This name is weird, but correct I suppose 😅

**Review by afonsojramos — APPROVED** [review]
Damn, this is one high quality pr ❤️‍🔥 Love it 💥

File: src/context/App.tsx (hunk 1 above)
Comment (setchy) [reply src/context/App.tsx:168]: Super weird.  I was trying to think _why_ we need this specific call at all...

Comment (afonsojramos) [reply src/context/App.tsx:168]: Considering that this lives under the validateToken, I assume that it does just that 😅 Testing the token with a simple and light request just to test the permissions I suppose. BUT, we could also do it the other way, and test the token directly by looking at the scopes, similar to how we were doing in the color scopes (?)

Comment (setchy) [reply src/context/App.tsx:168]: Future us problem haha

**Review by bmulholland — APPROVED** [review]
Agreed, this is beautiful, nice work!

**Comment by setchy** [comment]
Thanks gents 🙇

**Pull request merged**